src/main/java/org/opensearch/neuralsearch/sparse/cache/AbstractLRUCache.java

```diff
@@ -0,0 +1,145 @@
+import java.util.Set;
+
+/**
+ * Abstract LRU cache implementation for sparse vector caches.
+ * This class provides common functionality for managing eviction of cache entries
+ * based on least recently used policy.
+ *
+ * @param <LRUCacheKey> The type of key used for cache entries
+ */
+@Log4j2
+public abstract class AbstractLRUCache<LRUCacheKey> {
+    /**
+     * Map to track access with LRU ordering
+     */
+    protected final Set<LRUCacheKey> accessRecencySet;
+
+    /**
+     * The default initial capacity - MUST be a power of two.
+     */
+    static final int DEFAULT_INITIAL_CAPACITY = 1 << 4; // aka 16
+    /**
+     * The load factor used when none specified in constructor.
+     */
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    protected AbstractLRUCache() {
+        this.accessRecencySet = Collections.synchronizedSet(
+            Collections.newSetFromMap(new LinkedHashMap<>(DEFAULT_INITIAL_CAPACITY, DEFAULT_LOAD_FACTOR, true))
+        );
+    }
+
+    /**
+     * Updates access to an item for a specific cache key.
+     * This updates the item's position in the LRU order.
+     *
+     * @param key The key being accessed
+     */
+    protected void updateAccess(LRUCacheKey key) {
+        if (key == null) {
+            return;
+        }
+
+        accessRecencySet.add(key);
+    }
+
+    /**
+     * Retrieves the least recently used key without affecting its position in the access order.
+     *
+     * @return The least recently used key, or null if the cache is empty
+     */
+    protected LRUCacheKey getLeastRecentlyUsedItem() {
+        // With accessOrder is true in the LinkedHashMap, the first entry is the least recently used
+        Iterator<LRUCacheKey> iterator = accessRecencySet.iterator();
+        if (iterator.hasNext()) {
+            return iterator.next();
+        }
+        return null;
+    }
+
+    /**
+     * Evicts least recently used items from cache until the specified amount of RAM has been freed.
+     * For thread safety, please use synchronized when calling this method
+     *
+     * @param ramBytesToRelease Number of bytes to evict
+     */
+    public void evict(long ramBytesToRelease) {
+        if (ramBytesToRelease <= 0) {
+            return;
+        }
+
+        long ramBytesReleased = 0;
+
+        // Continue evicting until we've freed enough memory or the cache is empty
+        while (ramBytesReleased < ramBytesToRelease) {
+            // Get the least recently used item
+            LRUCacheKey leastRecentlyUsedKey = getLeastRecentlyUsedItem();
+
+            if (leastRecentlyUsedKey == null) {
+                // Cache is empty, nothing more to evict
+                break;
+            }
+
+            // Evict the item and track bytes freed
+            long bytesFreed = evictItem(leastRecentlyUsedKey);
+
+            if (bytesFreed > 0) {
+                ramBytesReleased += bytesFreed;
+                logEviction(leastRecentlyUsedKey, bytesFreed);
```

Comment:
feel unnecessary to put a debug log into a function especially as a function of an interface